### PR TITLE
feat: [PIE-5593]: PipelineChaining- added code source & view button to the pipeline selection modal

### PR DESCRIPTION
**Pull request by Eric761**
### Summary

Add code source and view button to the pipeline selection modal

#### Video

https://user-images.githubusercontent.com/73639897/194056760-d4cb9f80-55d7-44e2-9d4d-4a57eda3255d.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 10 Issues 
* Code issue 10 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11641)

**Comment by Eric761** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx [hunk 1]
```diff
@@ -48,29 +72,113 @@ const PipelineNameIdTagCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = (
...
+
+  return (
+    <div className={css.codeSourceColumnContainer}>
+      <Popover
+        disabled={!isRemote}
+        position={Position.TOP}
+        interactionKind={PopoverInteractionKind.HOVER}
+        className={Classes.DARK}
+        content={
+          <Layout.Vertical spacing="small" padding="large" style={{ maxWidth: 380 }}>
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:101]: css file

File: src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx [hunk 2]
```diff
@@ -48,29 +72,113 @@ const PipelineNameIdTagCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = (
...
+            {isRemote ? remoteWrapper.textName : inlineWrapper.textName}
+          </Text>
+        </div>
+      </Popover>
+    </div>
   )
+}
+
+// eslint-disable-next-line react/function-component-definition
+const ViewPipelineButtonCell: Renderer<CellProps<PMSPipelineSummaryResponse>> = ({ row, column }) => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:135]: nit: These functions can be moved to another file for columns/cells

File: src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx [hunk 3]
```diff
@@ -82,14 +190,22 @@ export function PipelineList({
...
+        Cell: CodeSourceCell
+      },
+      {
+        Header: getString('pipeline.openPipelineInNewTab'),
         accessor: 'modules',
-        width: '50%',
-        Cell: ModuleCell
+        width: '25%',
+        Cell: ViewPipelineButtonCell,
+        orgId,
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:207]: nit: Not sure about the implications completely, but naming can be in line with **orgIdentifier** and **projectIdentifier**

Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:207]: Currently, the usages of accountId and accountIdentifier is causing an additional assignment in certain cases

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx (hunk 2 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:135]: Sure

File: src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx (hunk 3 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/CommonPipelineStages/PipelineStage/PipelineList.tsx:207]: Here, naming can be in line with orgIdentifier and projectIdentifier.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**